platform/linuxbsd/detect.py

```diff
@@ -163,21 +174,19 @@ def configure(env):
             env.Append(CCFLAGS=["-fsanitize-recover=memory"])
             env.Append(LINKFLAGS=["-fsanitize=memory"])
 
-    if env["use_lto"]:
-        if not env["use_llvm"] and env.GetOption("num_jobs") > 1:
+    if env["lto"] != "none":
+        if env["lto"] == "thin":
+            if not env["use_llvm"] or env["linker"] != "lld":
```

Comment:
Is it only `lld`? `gold` seems to be supported as well - https://clang.llvm.org/docs/ThinLTO.html#linkers

Reply:
Good question! I'll test.

Reply:
> Is it only `lld`? `gold` seems to be supported as well - https://clang.llvm.org/docs/ThinLTO.html#linkers

`mold` also seems to support `-flto=thin` option https://github.com/rui314/mold/issues/367

Reply:
So from early testing it seems to work. I'll have to do more comparative tests between `-flto=thin` and `-flto` to see the actual gain. Testing LTO builds locally during a heat wave is not a good idea :P

Testing on my laptop, it takes a good 13 min to link with `gold`. I also found that it seems to work with GNU `ld` too!
It also seems to link in 13 min with `lld` so I guess it's working.

Here's the output for `gold` (`-v`):
```
 "/usr/bin/ld.gold" --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o bin/godot.linuxbsd.opt.tools.64.llvm /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crt1.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crti.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/crtbegin.o -L/usr/bin/../lib/gcc/x86_64-mageia-linux/12 -L/usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/lib -L/usr/lib -plugin /usr/bin/../lib64/LLVMgold.so -plugin-opt=mcpu=x86-64 -plugin-opt=thinlto platform/linuxbsd/godot_linuxbsd.linuxbsd.opt.tools.64.llvm.o [ ... snip ... ] modules/text_server_adv/libicu_builtin.linuxbsd.opt.tools.64.llvm.a -lXcursor -lXinerama -lXext -lXrandr -lXrender -lX11 -lXi -ldbus-1 -lGL -lpthread -ldl -l:libatomic.a -Bstatic -lstdc++ -Bdynamic -lm -lgcc -lgcc_eh -lc -lgcc -lgcc_eh /usr/bin/../lib/gcc/x86_64-mageia-linux/12/crtend.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crtn.o
 ```
 
And the output for GNU `ld`:
 ```
  "/usr/bin/ld" --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o bin/godot.linuxbsd.opt.tools.64.llvm /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crt1.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crti.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/crtbegin.o -L/usr/bin/../lib/gcc/x86_64-mageia-linux/12 -L/usr/bin/../lib/gcc/x86_64-mageia-linux/12/.././../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/lib -L/usr/lib -plugin /usr/bin/../lib64/LLVMgold.so -plugin-opt=mcpu=x86-64 -plugin-opt=thinlto platform/linuxbsd/godot_linuxbsd.linuxbsd.opt.tools.64.llvm.o [ ... snip ... ] modules/text_server_adv/libicu_builtin.linuxbsd.opt.tools.64.llvm.a -lXcursor -lXinerama -lXext -lXrandr -lXrender -lX11 -lXi -ldbus-1 -lGL -lpthread -ldl -l:libatomic.a -Bstatic -lstdc++ -Bdynamic -lm -lgcc -lgcc_eh -lc -lgcc -lgcc_eh /usr/bin/../lib/gcc/x86_64-mageia-linux/12/crtend.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crtn.o
```

And finally output with `lld`:
```
 "/usr/bin/ld.lld" --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o bin/godot.linuxbsd.opt.tools.64.llvm /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crt1.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crti.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/crtbegin.o -L/usr/bin/../lib/gcc/x86_64-mageia-linux/12 -L/usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/lib -L/usr/lib -plugin-opt=mcpu=x86-64 -plugin-opt=thinlto platform/linuxbsd/godot_linuxbsd.linuxbsd.opt.tools.64.llvm.o [ ... snip ... ] modules/text_server_adv/libicu_builtin.linuxbsd.opt.tools.64.llvm.a -lXcursor -lXinerama -lXext -lXrandr -lXrender -lX11 -lXi -ldbus-1 -lGL -lpthread -ldl -l:libatomic.a -Bstatic -lstdc++ -Bdynamic -lm -lgcc -lgcc_eh -lc -lgcc -lgcc_eh /usr/bin/../lib/gcc/x86_64-mageia-linux/12/crtend.o /usr/bin/../lib/gcc/x86_64-mageia-linux/12/../../../../lib64/crtn.o
```

Reply:
I removed the checks so now `lto=thin` can be used with any linker, since they all seem to support it.

We'd need to benchmark both build times and performance to see which one should be recommended for each platform/compiler/linker combination.